var/spack/repos/builtin/packages/py-pyprecice/package.py

```diff
@@ -14,15 +14,18 @@ class PyPyprecice(PythonPackage):
 
     homepage = "https://precice.org"
     git = "https://github.com/precice/python-bindings.git"
-    url = "https://github.com/precice/python-bindings/archive/v2.4.0.0.tar.gz"
+    url = "https://github.com/precice/python-bindings/archive/v3.1.1.tar.gz"
     maintainers("ajaust", "BenjaminRodenberg", "IshaanDesai")
 
-    license("LGPL-3.0")
-
     # Always prefer final version of release candidate
     version("develop", branch="develop")
+    version("3.1.1", sha256="50a0f1cbdb8fc362c22d316151c0e757958ff136a094e63b9b82d045d01d19c7")
+    version("3.1.0", sha256="8d9bd9e28859001ab503a1e2f90e54b3c000079f04c14dc7c0c04c61c5666641")
+    version("3.0.0.0", sha256="7e2c4b106a231b0df2a430d86d4a7b295f85adbe3478c425f863d1a4bebee9f7")
+    version("2.5.0.4", sha256="7f9449573eb52ce48ca3f0ab35529ea0064942487842515ae0a2c9299aa0f0db")
+    version("2.5.0.3", sha256="b983229b9fdf6bd4605ae8710985eb681025f6fb28ad8d7736cdf92593eef6df")
     version("2.5.0.2", sha256="6d7b78da830db6c5133b44617196ee90be8c7d6c8e14c8994a4800b3d4856416")
-    version("2.5.0.1", sha256="e2602f828d4f907ea93e34f7d4adb8db086044a75a446592a4099423d56ed62c")
+    version("2.5.0.1", sha256="d7c666e6ebff9e007c3703d8e3c3fcdf0f45289e36c2c17223b3aedc3259ab6c")
```

Comment:
We did not patch the release. As described below we did a post-tag bump directly after the release to keep the checksum stable. See https://github.com/precice/python-bindings/commit/0d28cf00989cc04e19eb12d405d4379ce11744e4 and it's parent release commit https://github.com/precice/python-bindings/commit/fcee5ffe97a1660c6e67afdd03610efa3b6024b2. We updated the `package.py` file a few days later via https://github.com/spack/spack/pull/32551. Therefore, I'm a bit surprised that the checksum is not correct (anymore?). 

To answer your question: I think it has almost always been wrong since I did not check the checksum for quite some time. We should definitely keep an eye on the checksum of 3.1.1. If it changes again there are some factors we are currently not aware of.

I developed a bit of CI in our repository and I think we could include `./spack checksum --verify -l py-pyprecice` to regularly verify the checksum of the latest version and spot errors early.